### PR TITLE
Several fixes for three-operand conversion

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Signed-off-by: Mihai Budiu <mbudiu@vmware.com>

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
@Kixiron this fixes several bugs in the three-operand conversion, but not all.
I have thus marked the PR as a draft.
But you should try this, since it should give you useful results for many queries.

**Comment by Kixiron** [comment]
This pr makes the body of the function get put into a block within function calls a la
```rust
let x = wrap_bool({
    let a = ...;
    let b = ...;
    b
});
x
```

**Comment by mbudiu-vmw** [comment]
@Kixiron I hope now I have fixed the three-operand code generation.
I am also testing it thoroughly be running the three-operand Rust code. The Rust compiler likes the result.

File: SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/ThreeOperandVisitor.java [hunk 1]
```diff
@@ -47,6 +47,36 @@
  * };
  */
 public class ThreeOperandVisitor extends InnerVisitor implements Function<IDBSPInnerNode, IDBSPInnerNode>, IModule {
+    static class PendingStatements {
```
Comment (lalithsuresh) [inline SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/ThreeOperandVisitor.java:50]: This sounds like it should be called a block?

File: SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/ThreeOperandVisitor.java [hunk 2]
```diff
@@ -112,27 +139,28 @@ public boolean preorder(DBSPExpression expression) {
...
         return stat.getVarReference();
     }
 
+    /**
+     * Generate a block from all the statements in the last pending list
+     * and return an expression equivalent to 'last'.
+     */
     DBSPExpression makeBlock(@Nullable DBSPExpression last) {
-        if (this.toInsert.isEmpty())
+        List<DBSPStatement> toInsert = this.pending.getCurrentAndClear();
```
Comment (lalithsuresh) [inline SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/ThreeOperandVisitor.java:160]: makeBlock should be a method of `PendingStatement` (better called `Block`).

File: SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/ThreeOperandVisitor.java (hunk 1 above)
Comment (mbudiu-vmw) [reply SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/ThreeOperandVisitor.java:50]: It will eventually become the contents of a block.

**Review by lalithsuresh — APPROVED** [review]

**Pull request merged**